src/GeniusYield/GYConfig.hs

```diff
@@ -72,6 +77,33 @@ data GYCoreProviderInfo
   | GYBlockfrost {cpiBlockfrostKey :: !(Confidential Text)}
   deriving stock (Show)
 
+coreProviderIO :: FilePath -> IO GYCoreProviderInfo
+coreProviderIO filePath = do
+  bs <- LBS.readFile filePath
+  case Aeson.eitherDecode' bs of
+    Left err  -> throwIO $ userError err
+    Right cfg -> pure cfg
+
+isNodeChainIx :: GYCoreProviderInfo -> Bool
+isNodeChainIx GYNodeChainIx{} = True
+isNodeChainIx _               = False
+
+
+isDbSync :: GYCoreProviderInfo -> Bool
+isDbSync GYDbSync{} = True
+isDbSync _          = False
+
+isMaestro :: GYCoreProviderInfo -> Bool
+isMaestro GYMaestro{} = True
+isMaestro _           = False
+
+findProviderToken :: String -> IO Text
```

Comment:
`findMaestroProviderToken`

Reply:
@TomasOsiecki Do you plan to add that case?

Reply:
Finally i decided to use `findMaestroProviderToken` because with the changes from main, now we have a list of configurations (maestro and blockfrost) and we use the function `isMaestro` to find the configuration.